src/components/simple-color-picker/simple-color-picker.test.js

```diff
@@ -0,0 +1,473 @@
+          expect(position.top).to.be.greaterThan(topLess);
+          expect(position.left).to.be.lessThan(leftLess + additionVal);
+          expect(position.left).to.be.greaterThan(leftLess);
+        });
+      }
+    );
+
+    it("should render SimpleColorPicker with required prop", () => {
+      CypressMountWithProviders(<SimpleColorPickerCustom required />);
+
+      verifyRequiredAsteriskForLegend();
+    });
+
+    it.each(validationTypes)(
+      "should render SimpleColorPicker and set type to %s and set as string",
+      (type, color) => {
+        CypressMountWithProviders(
+          <SimpleColorPickerCustom {...{ [type]: "Message" }} />
+        );
+
+        simpleColorPickerComponent()
+          .find("div")
+          .find(`[data-element="${type}"]`)
+          .trigger("mouseover");
+        getDataElementByValue("tooltip").should("have.text", "Message");
+        simpleColorPickerComponent()
+          .find("div")
+          .children()
+          .should("have.css", "outline-color", color);
+        verifyBeforeColor(type, color);
+      }
+    );
+
+    it.each(validationTypes)(
+      "should render SimpleColorPicker and set type to %s as string and have validationOnLegend prop",
+      (type, color) => {
+        CypressMountWithProviders(
+          <SimpleColorPickerCustom
+            {...{ [type]: "Message" }}
+            validationOnLegend
+          />
+        );
+
+        simpleColorPickerComponent()
+          .find("legend")
+          .find(`[data-element="${type}"]`)
+          .trigger("mouseover");
+        getDataElementByValue("tooltip").should("have.text", "Message");
+        simpleColorPickerComponent()
+          .find("div")
+          .children()
+          .should("have.css", "outline-color", color);
+        verifyBeforeColor(type, color);
+      }
+    );
+
+    it.each(validationTypes)(
+      "should render SimpleColorPicker and set type to %s as boolean",
+      (type, color) => {
+        CypressMountWithProviders(
+          <SimpleColorPickerCustom {...{ [type]: true }} />
+        );
+
+        simpleColorPickerComponent()
+          .find(`[data-element="${type}"]`)
+          .should("not.exist");
+        simpleColorPickerComponent()
+          .find("div")
+          .children()
+          .should("have.css", "outline-color", color);
+      }
+    );
+  });
+
+  describe("should render SimpleColorPicker component and check events", () => {
+    let callback;
+
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it("should call onChange callback when a click event is triggered", () => {
+      CypressMountWithProviders(
+        <SimpleColorPickerCustom onChange={callback} />
+      );
+
+      simpleColorPickerInput(5)
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it.each([
```

Comment:
I think we can write these tests to assert that a relevant `SimpleColor` square is focussed and that the callback stub is called with the relevant colour. We should do this for `right`/ `left` `up` and `down` imo

```
const indexes = Array.from({ length: colors.length }).map((_, index) => index);

it("should call onChange callback and focus the correct item when the right arrow key is triggered", () => {
    CypressMountWithProviders(
          <SimpleColorPickerCustom onChange={callback} />
        );
        simpleColorPickerInput(0).click()

        indexes.forEach((index) => {
          const next = index < colors.length - 1 ? index + 1 : 0;

          simpleColorPickerInput(index)
          .trigger("keydown", keyCode("rightarrow"))
          .then(() => {
            // eslint-disable-next-line no-unused-expressions
            expect(callback).to.have.been.calledWith(colors[next].color);
          })
          .then(() => {
              simpleColorPickerInput(next).should("be.focused")
            });
        })
      }
    );

    it("should call onChange callback and focus the correct item when the left arrow key is triggered", () => {
      CypressMountWithProviders(
            <SimpleColorPickerCustom onChange={callback} />
          );
          simpleColorPickerInput(0).click()
  
          indexes.reverse().forEach((index) => {
            const next = index > 0 ? index - 1 : colors.length - 1;
  
            simpleColorPickerInput(index)
            .trigger("keydown", keyCode("leftarrow"))
            .then(() => {
              // eslint-disable-next-line no-unused-expressions
              expect(callback).to.have.been.calledWith(colors[next].color);
            })
            .then(() => {
                simpleColorPickerInput(next).should("be.focused")
              });
          })
        }
      );
  ```

Reply:
thx :) applied